aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -56,19 +56,20 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
             # installed the dependencies into artifact folder
             self.actions.append(NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm))
 
-            # if dependencies folder exists, copy dependencies into dependencies into dependencies folder
+            # if dependencies folder exists, Move dependencies from artifact folder to dependencies folder
             if self.dependencies_dir:
-                self.actions.append(CopyDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
+                self.actions.append(MoveDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
+
+        # if dependencies folder exists and not download dependencies, simply copy the dependencies from the
+        # dependencies folder to artifact folder
+        # if combine_dependencies is false, will not copy the dependencies from dependencies folder to artifact folder
+        if self.dependencies_dir and self.combine_dependencies:
+            self.actions.append(CopySourceAction(self.dependencies_dir, artifacts_dir))
```

Comment:
If we have download_dependencies True, dependencies_dir not None, and combine_dependencies True. The workflow will install deps into artifacts_dir, move the deps into dependencies_dir, and then copy the deps back into artifacts folder? Can we make the workflow smarter so that we can minimize the amount of IO?

Reply:
Updated it with a more efficient solution, but the code will looks a little bit complicated.